deepspeed/runtime/engine.py

```diff
@@ -496,8 +496,9 @@ def _configure_with_arguments(self, args, mpu):
         # environment variable is set. We must align args.local_rank to this value for
         # backwards compatability with scripts relying on [args|self].local_rank containing
         # the correct local rank info.
-        args.local_rank = int(os.environ['LOCAL_RANK'])
-        self.local_rank = args.local_rank
+        self.local_rank = int(os.environ['LOCAL_RANK'])
```

Comment:
Could we add a sanity check that 'LOCAL_RANK' is in os.environ?

Reply:
`int(os.environ.get('LOCAL_RANK', '-1'))` ?

Reply:
If we set local rank to -1 in this section of the code we'll attempt to send the model to device -1 which would error out later. I'll add an assert with a useful error message here.

Reply:
well, I only meant that -1 can be a default for no-local-rank, which is what we use and of course check later - but you're right deepspeed always requires `local_rank> -1`  so the fallback I proposed makes no sense.